tests/test_markdown.py

```diff
@@ -0,0 +1,86 @@
+import io
+import os
+import textwrap
+
+from readme_renderer.markdown import render
+
+
+def test_simple():
+    markdown_markup = 'Hello'
+    out = render(markdown_markup)
+    assert out == '<p>Hello</p>'
+
+
+def test_url_no_link():
+    markdown_markup = 'http://mymalicioussite.com/'
+    out = render(markdown_markup)
+    expected_html = '<p>http://mymalicioussite.com/</p>'
+    assert out == expected_html
+
+
+def test_iframe():
+    markdown_markup = """\
+        <iframe src="http://mymalicioussite.com/">Click here</iframe>
+    """.strip()
+    out = render(markdown_markup)
+    expected_html = ''.join([
+        '&lt;iframe src="http://mymalicioussite.com/"&gt;'
+        'Click here&lt;/iframe&gt;'])
+    assert out == expected_html
+
+
+def test_script():
+    markdown_markup = textwrap.dedent("""\
+        <script>
+            alert("Hello");
+        </script>""")
+    out = render(markdown_markup)
+    expected_html = textwrap.dedent("""\
+        &lt;script&gt;
+            alert("Hello");
+        &lt;/script&gt;""")
+    assert out == expected_html
+
+
+def test_a_tag_gets_nofollow():
+    markdown_markup = '<a href="http://mymalicioussite.com/">Click here</a>'
+    out = render(markdown_markup)
+    expected_htmls = [
+        ''.join(['<p><a rel="nofollow" href="http://mymalicioussite.com/">',
+                 'Click here</a></p>']),
+        ''.join(['<p><a href="http://mymalicioussite.com/" rel="nofollow">',
+                 'Click here</a></p>']),
+    ]
+    assert out in expected_htmls
+
+
+def test_smart_strong():
+    markdown_markup = 'Text with double__underscore__words.'
+    out = render(markdown_markup)
+    expected_html = '<p>Text with double__underscore__words.</p>'
+    assert out == expected_html
+
+
+def test_github_backtick_syntax_for_code():
+    _do_test_with_files('code')
+
+
+def test_headings_and_paragraphs():
+    _do_test_with_files('headings_and_paragraphs')
+
+
+def test_misc():
+    _do_test_with_files('misc')
+
+
+def _do_test_with_files(test_name):
+    md_markup = read('{0}.md'.format(test_name))
+    expected_html = read('{0}.html'.format(test_name))
+    out = render(md_markup)
+    assert out == expected_html.rstrip()
+
+
+def read(fn):
+    path = os.path.join(os.path.dirname(__file__), 'fixtures', 'markdown', fn)
+    with io.open(path, encoding='utf-8') as f:
```

Comment:
Yes, I'm OK to change it later when python2 compat is dropped.

Look like pypi-legacy has an indeterminate lifespan, but if we pin a legacy version of readme_renderer in pypi-lecacy requirements, we'll be able to drop python2 compatibility. For the moment, I think it's better to keep with python2 compatibility so pypi-legacy can at least get markdown.